src/modules/roc_fec/composer.h

```diff
@@ -160,7 +160,7 @@ class Composer : public packet::IComposer, public core::NonCopyable<> {
         PayloadID& payload_id = *(PayloadID*)fec.payload_id.data();
 
         payload_id.clear();
-        payload_id.set_sbn(fec.blknum);
+        payload_id.set_sbn(fec.source_block_number);
```

Comment:
I would add a check:

```
        roc_panic_if(fec.source_block_number >= (uint32_t)-1);
        payload_id.set_sbn((uint32_t)fec.source_block_number);
```

Or maybe:

```
        roc_panic_if(sizeof(blknum_t) != sizeof(uint32_t));
        payload_id.set_sbn(fec.source_block_number);
```

To ensure that we don't lose bits here.